File: Product Array/product_array.java
class product_array
{
void product(int arr[] , int n)
{
	int i,temp=1;
	int prod[] = new int[n];
	if(n == 1)
	{
		System.out.println("0");
		return;
	}
	for(int j=0;j<n;j++)
	{
		prod[j] = 1;
	}
	for(i=0;i<n;i++)
	{
		prod[i] = temp;
		temp *= arr[i];
	}
	temp = 1;
	for(i=n-1;i>=0;i--)
	{
		prod[i] *= temp;
		temp *= arr[i];
	}
	for(i=0;i<n;i++)
	{
		System.out.print(prod[i] + " ");
	}
	return;
}
public static void main(String[]args)
{
product_array p = new product_array();
int arr[] = {1,2,3,4,5};
int n = arr.length;
System.out.println("Array :- ");
p.product(arr,n);
}
}

File: Product Array/product_array.py
def product_array(arr,n):
    if(n == 1):
        print(0)
        return
    i , temp = 1 , 1
    prod = [1 for i in range(n)]
    for i in range(n):
        prod[i] = temp
        temp = temp * arr[i]
    temp = 1
    for i in range(n-1,-1,-1):
        prod[i] = prod[i]*temp
        temp = temp * arr[i]
    for i in range(n):
        print(prod[i] , end = " ")
    return
    
arr = [1,2,3,4,5]
n = len(arr)
print("Product Array :- ")
product_array(arr,n)
     
